window-purpose-utils.el

```diff
@@ -105,9 +105,7 @@ Example:
 for each entry in hash-table TABLE."
   (let (results)
     (maphash #'(lambda (key value)
-                 (setq results
-                       (append results
-                               (list (funcall function key value)))))
+                 (push (funcall function key value) results))
```

Comment:
Looks good. Do you think `(setq results (cons (funcall function key value) results))` will have better performance than `push` ? I'd expect `push` to use `cons` internally anyway, but the difference in performance is probably negligible.

Reply:
```lisp
(push (funcall ...) results)
```
should macroexpand to 
```lisp
(setq results (cons (funcall ...) results))
```
so I expect the performance to be identical (except slightly longer compilation time since `push` needs to macroexpand)